### PR TITLE
Add splunkhttp module and release automation

### PR DESCRIPTION
**Pull request by pellared**
## Why

Per https://github.com/signalfx/splunk-otel-go/pull/18#discussion_r602250964

## What

- Add `splunkhttp` module.
- Add automation for a multi-module repository.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -73,28 +66,71 @@ test-verbose: ARGS=-v
...
+# Pre-release targets
+
+.PHONY: add-tag
+add-tag: # example usage: make add-tag tag=v1.100.1
+	$Q [ "$(tag)" ] || ( echo ">> 'tag' is not set"; exit 1 )
+	@echo "Adding tag $(tag)"
+	$Q git tag -a $(tag) -m "Version $(tag)"
+	$Q set -e; for dir in $(SUBMODULES); do \
+	  (echo Adding tag "$${dir:2}/$(tag)" && \
+	 	git tag -a "$${dir:2}/$(tag)" -m "Version ${dir:2}/$(tag)" ); \
```
[the remark this answers is not in the document]
Comment (pellared) [reply Makefile]: @MrAlias 
Did it here: https://github.com/signalfx/splunk-otel-go/pull/27/commits/cd711a6a9089f44bf7b307e119675d3b7dc2af73
I saw that both you and @owais are signing your commits 👍 
I am OK to require signed commits in `main` branch as well as "use GitHub auth to restrict releasing" (not sure what it exactly means TBH 😄 )
Also, I propose to set only "Squash merge" as the way PRs are getting merged. I see it is the way OTel does it: https://github.com/open-telemetry/opentelemetry-go/commits/main . I also like it more than rebasing or "normal" merging.

File: Makefile [hunk 2]
```diff
@@ -73,28 +66,72 @@ test-verbose: ARGS=-v
...
+	$Q set -e; for dir in $(SUBMODULES); do \
+	  (echo Deleting tag "$${dir:2}/$(tag)" && \
+	 	git tag -d "$${dir:2}/$(tag)" ); \
+	done
+
+.PHONY: push-tag
+push-tag: # example usage: make push-tag remote=origin tag=v1.100.1 commit=<hash>
+	$Q [ "$(remote)" ] || ( echo ">> 'remote' is not set"; exit 1 )
+	$Q [ "$(tag)" ] || ( echo ">> 'tag' is not set"; exit 1 )
+	$Q [ "$(commit)" ] || ( echo ">> 'commit' is not set"; exit 1 )
```
Comment (MrAlias) [inline Makefile:110]: This looks like it was added in the wrong target. I was thinking specifying the commit explicitly in `add-tag`. I'm pretty sure having it here will not only push the tag but try to push that commit in addition.

Comment (pellared) [reply Makefile:110]: You are correct. I am a dummy :D

Comment (pellared) [reply Makefile]: Addressed c64ba7d3

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+Failure to do so will leave things in a broken state.
+
+***IMPORTANT***: [There is currently no way to remove an incorrectly tagged version of a Go module](https://github.com/golang/go/issues/34189).
+It is critical you make sure the version you push upstream is correct.
+[Failure to do so will lead to minor emergencies and tough to work around](https://github.com/open-telemetry/opentelemetry-go/issues/331).
+
+1. Run on the main branch for the merged Pull Request.
+
+    ```sh
+    make add-tag tag=<new tag>
```
Comment (MrAlias) [inline RELEASING.md:39]:
```suggestion
1. Run on the main branch and specify the commit for the merged Pull Request.

    ```sh
    make add-tag tag=<new tag> commit=<commit>
```

**Pull request merged**